_posts/2020-09-07-introduction-to-system-tests.md

```diff
@@ -0,0 +1,199 @@
+---
+layout: post
+title:  "Introduction to system tests"
+date: 2020-09-07
+author: maroš_orsák
+---
+
+Lessons learn from past years shown that testing is the essential part of the software cycle. We have known many 
+failures that happened because of neglect testing. In Strimzi, we exactly know how we should test the features. 
+Follow me on this blog post and see the basic view of our tests.
+
+In this blog post we are gonna look at system tests and I will give you motivation why you should look at it. The content 
+is as follows:
+
+# Content
+1. [Motivation](#Resources)
+2. [How they differ from our UT and IT](#Lifecycle of tests)
+3. [How they run in relation with Kubernetes](#Auxiliary classes)
+4. [How to run them](#How to create a system test)
+5. [Conclusion](#Conclusion)
+
+## Motivation
+
+Maybe you ask yourself these questions. Why do we need to write system tests? What is the value of it? Do they catch
+errors that unit or integration tests doesn't? The answer is very simple. Yes! `System tests` are the core unit to be able ship
+the product to the customer with more confidence that it will not crash in the production. These types of tests, which
+validate the basic properties of system we call [Smoke tests](https://www.guru99.com/smoke-testing.html). They ensure to detect
+major issues such as deployment of application with general configuration and so on. Moreover, we have got the regression
+test suites, which coverage most of all features and will definitely find a bug. In the next section i will describe the
+difference between our system tests, integration and unit tests.
+
+## How they differ from our UT and IT
+
+Firstly, we should know what is the purpose of the Unit test. Let's define information definition of unit test.
+
+> A unit test exercises a single behavior of a software module. That module is usually a class, and the behavior is usually a public method of the class. The test asserts that the actual result matches the expected result. These assertions must all pass, or the unit test will fail. (Ryan Cook)
+
+Our unit tests doesn't do anything special. For clarification I can show you an example of our unit test for instance in `cluster
+operator` module. This is the test case, which verify that if we setup `Kafka` with external listener Nodeport then 
```

Comment:
I think if you format it as code it should be `cluster-operator` as the name of the Maven module. Or you should just write it as _Cluster Operator_ without the code formatting. Similarly, Kafka should be just Kafka and not Kafka`

Reply:
I will use the first option with the maven module name.